extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/RedpandaKafkaContainer.java

```diff
@@ -31,12 +31,16 @@ final class RedpandaKafkaContainer extends GenericContainer<RedpandaKafkaContain
     private static final int PANDAPROXY_PORT = 8082;
 
     RedpandaKafkaContainer(DockerImageName dockerImageName, int fixedExposedPort, String defaultNetworkId,
-            boolean useSharedNetwork, RedpandaBuildTimeConfig redpandaConfig) {
+            String serviceName, boolean useSharedNetwork, RedpandaBuildTimeConfig redpandaConfig) {
         super(dockerImageName);
         this.fixedExposedPort = fixedExposedPort;
         this.useSharedNetwork = useSharedNetwork;
         this.redpandaConfig = redpandaConfig;
 
+        if (serviceName != null) { // Only adds the label in dev mode.
+            withLabel(DevServicesKafkaProcessor.DEV_SERVICE_LABEL, serviceName);
+        }
+
```

Comment:
As above - I feel like this may be a weak way of checking for what we're interested in, because if we started labelling test mode containers for some other reason, we'd do the wrong thing here.